pydantic_settings/main.py

```diff
@@ -1,13 +1,10 @@
 from __future__ import annotations as _annotations
 
-import warnings
-from typing import AbstractSet, Any, ClassVar, Dict, List, Optional, Tuple, Type, Union
+from typing import Any, Dict, Optional, Tuple, Type
 
-from pydantic.config import BaseConfig, Extra
-from pydantic.fields import ModelField
+from pydantic import ConfigDict, DirectoryPath, FilePath
+from pydantic._internal._utils import deep_update
```

Comment:
Are we okay using pydantic internals here? I think it's fine since it's still our package, just want to make sure.

(If we do this, and I guess even if we don't, it might be a good idea to run the tests for pydantic_settings in the CI for pydantic.)

Reply:
> Are we okay using pydantic internals here? I think it's fine since it's still our package, just want to make sure.

Yes, as we maintain Pydantic I think it's fine

> (If we do this, and I guess even if we don't, it might be a good idea to run the tests for pydantic_settings in the CI for pydantic.)

Agree